Modules/Filtering/ImageGrid/test/itkResampleImageTest7.cxx

```diff
@@ -0,0 +1,170 @@
+    value = index[0] + index[1];
+    iter.Set(value);
+    }
+  std::cout << "Done." << std::endl;
+
+  // Create an affine transformation
+  AffineTransformType::Pointer affineTransform = AffineTransformType::New();
+  affineTransform->Scale(2.0);
+
+  // Create and configure a resampling filter
+  typedef itk::ResampleImageFilter< ImageType, ImageType > ResampleFilterType;
+  ResampleFilterType::Pointer resample = ResampleFilterType::New();
+
+  typedef itk::PipelineMonitorImageFilter< ImageType > MonitorFilter;
+  MonitorFilter::Pointer monitor = MonitorFilter::New();
+
+  typedef itk::StreamingImageFilter<ImageType,ImageType> StreamerType;
+  StreamerType::Pointer streamer = StreamerType::New();
+
+  std::cout << "Test with normal AffineTransform." << std::endl;
+  resample->SetInput( image );
+  resample->SetTransform( affineTransform );
+  monitor->SetInput( resample->GetOutput() );
+  streamer->SetInput( monitor->GetOutput() );
+
+  unsigned char numStreamDiv;
+  
+  // Run the resampling filter without streaming, i.e. 1 StreamDivisions
+  numStreamDiv= 1; // do not split, i.e. do not stream
+  try
+    {
+    streamer->SetNumberOfStreamDivisions(numStreamDiv);
+    streamer->Update();
+    }
+  catch( itk::ExceptionObject & excp )
+    {
+    std::cerr << excp << std::endl;
+    return EXIT_FAILURE;
+    }
+  
+  if (!monitor->VerifyAllInputCanStream(numStreamDiv))
+    {
+    std::cout << "Avoiding streaming failed to execute as expected!" << std::endl;
+    std::cout << monitor;
+    return EXIT_FAILURE;
+    }
+
+  ImagePointerType outputNoSDI= streamer->GetOutput();
+  outputNoSDI->DisconnectPipeline();
+
+  // Run the resampling filter with streaming
+  numStreamDiv= 8; // split into numStream pieces for streaming.
+  resample->Modified();
+  try
+    {
+    streamer->SetNumberOfStreamDivisions(numStreamDiv);
+    streamer->Update();
+    }
+  catch( itk::ExceptionObject & excp )
+    {
+    std::cerr << excp << std::endl;
+    return EXIT_FAILURE;
+    }
+  
+  if (!monitor->VerifyAllInputCanStream(numStreamDiv))
+    {
+    std::cout << "Streaming failed to execute as expected!" << std::endl;
+    std::cout << monitor;
+    return EXIT_FAILURE;
+    }
+
+  ImagePointerType outputSDI= streamer->GetOutput();
+  outputSDI->DisconnectPipeline();
+
+  itk::ImageRegionIterator<ImageType>
+      itNoSDI(outputNoSDI, outputNoSDI->GetLargestPossibleRegion()),
+      itSDI(outputSDI, outputSDI->GetLargestPossibleRegion());
+  for(itNoSDI.GoToBegin(), itSDI.GoToBegin();
+      !itNoSDI.IsAtEnd() && !itSDI.IsAtEnd();
+      ++itNoSDI, ++itSDI)
+    {
+    if(itk::Math::NotAlmostEquals( itNoSDI.Value(), itSDI.Value() ))
+      {
+      std::cout << "Pixels differ "
+		<< itNoSDI.Value() << " "
+                << itSDI.Value() << std::endl;
+      return EXIT_FAILURE;
+      }
+    }
+  if(itNoSDI.IsAtEnd() != itSDI.IsAtEnd())
+    {
+    std::cout << "Iterators don't agree on end of image" << std::endl;
+    return EXIT_FAILURE;
+    }
```

Comment:
May be add a message saying that the test failed explicitly, as recommended in `C.23.2 Regressions in Tests` of the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf).